File: 项目文件/backend/app/schemas/note.py
"""笔记 Schema"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NoteCreate(BaseModel):
    title: str
    content: str | None = None
    category: str | None = None
    tags: list[str] | None = None
    is_pinned: bool = False
    parent_id: uuid.UUID | None = None


class NoteUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    category: str | None = None
    tags: list[str] | None = None
    is_pinned: bool | None = None
    parent_id: uuid.UUID | None = None


class NoteResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str
    content: str | None
    category: str | None
    tags: list[str] | None
    is_pinned: bool
    parent_id: uuid.UUID | None
    owner_id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class NoteListResponse(BaseModel):
    items: list[NoteResponse]
    total: int

File: 项目文件/frontend/src/types/note.ts
import type { Visibility } from '../utils/visibility';

export interface Note {
  id: string;
  title: string;
  content: string | null;
  category: string | null;
  tags: string[] | null;
  is_pinned: boolean;
  parent_id: string | null;
  owner_id: string;
  visibility?: Visibility;
  restricted_users?: string[] | null;
  created_at: string;
  updated_at: string;
}

export interface NoteCreate {
  title: string;
  content?: string | undefined;
  category?: string | undefined;
  tags?: string[] | undefined;
  is_pinned?: boolean | undefined;
  parent_id?: string | null | undefined;
  visibility?: Visibility | undefined;
  restricted_users?: string[] | undefined;
}

export interface NoteUpdate {
  title?: string | undefined;
  content?: string | undefined;
  category?: string | undefined;
  tags?: string[] | undefined;
  is_pinned?: boolean | undefined;
  parent_id?: string | null | undefined;
  visibility?: Visibility | undefined;
  restricted_users?: string[] | undefined;
}

export interface NoteListResponse {
  items: Note[];
  total: number;
}
